src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -181,27 +184,49 @@ object ProtoTypes {
 
     def argsAreTyped: Boolean = myTypedArgs.size == args.length
 
+    private def typedArg(arg: untpd.Tree, typerFn: untpd.Tree => Tree)(implicit ctx: Context): Tree = {
```

Comment:
I would also consider renaming it because having two methods called `typedArg` with the same number of arguments can be confusing